### PR TITLE
One page design for telemetry to load secret.

### PR DESCRIPTION
**Pull request by FFFFFaraway**
**What this PR does / why we need it**:
Add the design docs for telemetry to load secret instead of deviceshiifu.

**Will this PR make the community happier**? 

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**How is this PR tested**
- [ ] unit test
- [ ] e2e test
- [x] other (please specify)

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: docs/design/shifu-framework/telemetry-secret.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
+# Telemetry Secret Loading Design
+
+## Introduction
+
+Telemetry Service is a standalone service, hereafter referred to as TSCenter. But the settings in TelemetryService CRD is loaded by Deviceshifu and sent to TSCenter. If the setting contains password, there will exist the problem of plaintext password in communication between Deviceshifu and TSCenter.
```
Comment (kris21he) [inline docs/design/shifu-framework/telemetry-secret.md:5]: let's call this as telemetry service always and be consist across the doc. avoid introduce new concept like "TSCenter"

File: docs/design/shifu-framework/telemetry-secret.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+There could be multiple ways to deal with it:
```
Comment (kris21he) [inline docs/design/shifu-framework/telemetry-secret.md:7]: try make a table to list the methods and its pros & cons?

File: docs/design/shifu-framework/telemetry-secret.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+1. Add a custom encode/decode algorithm in communication between Deviceshifu and TSCenter.
+2. Let the TSCenter to manage all TelemetryService CRD info. Deviceshifu only specifies the name of TelemetryService CRD. But the TSCenter's workload may be too high.
+3. Deviceshifu still send all info as before but hide the password, instead, it will let the TSCenter to find the Secret by the name of TelemetryService CRD.
+
+We will use the last way here. There may still exist some improvement space, like:
```
Comment (kris21he) [inline docs/design/shifu-framework/telemetry-secret.md:13]: make a conclusion & future improvement sections here after the comparison of three methods?

File: docs/design/shifu-framework/telemetry-secret.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+1. Every time we push data, TSCenter needs to acquire the Secret.
+
+## General Design
+
+1. TSCenter created with k8s client.
```
Comment (kris21he) [inline docs/design/shifu-framework/telemetry-secret.md:19]: can we draw a diagram?

File: docs/design/shifu-framework/telemetry-secret.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+## Introduction
+
+Telemetry Service is a standalone service, hereafter referred to as TSCenter. But the settings in TelemetryService CRD is loaded by Deviceshifu and sent to TSCenter. If the setting contains password, there will exist the problem of plaintext password in communication between Deviceshifu and TSCenter.
+
+There could be multiple ways to deal with it:
+
+1. Add a custom encode/decode algorithm in communication between Deviceshifu and TSCenter.
+2. Let the TSCenter to manage all TelemetryService CRD info. Deviceshifu only specifies the name of TelemetryService CRD. But the TSCenter's workload may be too high.
+3. Deviceshifu still send all info as before but hide the password, instead, it will let the TSCenter to find the Secret by the name of TelemetryService CRD.
```
Comment (kris21he) [inline docs/design/shifu-framework/telemetry-secret.md:11]: let's use credential and be consist in the document for the words "password" "secret"

**Review by kris21he — APPROVED** [review]
LGTM

File: docs/design/shifu-framework/telemetry-secret.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+```
+
+1. Telemetry Service created with k8s client.
+2. Deviceshifu loads info from `TelemetryServiceCRD` and sends it to Telemetry Service, with the credential field to be the name of `Secret` that contains the base64-encoded credential.
+3. Telemetry Service load `Secret` and write credential into the setting.
+4. Grant Telemetry Service-sa with `Secret` read RBAC.
+
+## Design Details
+
+Currently, only SQLSetting uses the credential.
```
Comment (BtXin) [inline docs/design/shifu-framework/telemetry-secret.md:46]: MQTT and HTTP also has password/sercrets

**Review by tomqin93 — APPROVED** [review]
LGTM

**Pull request merged**